easybuild/easyblocks/g/gromacs.py

```diff
@@ -324,103 +384,91 @@ def configure_step(self):
-                                     self.cfg['mpi_numprocs'], self.cfg['parallel'])
-
-                self.cfg.update('configopts', "-DGMX_MPI=ON -DGMX_THREAD_MPI=OFF")
-
-                mpiexec = which(self.cfg['mpiexec'])
-                if mpiexec:
-                    self.cfg.update('configopts', "-DMPIEXEC=%s" % mpiexec)
-                    self.cfg.update('configopts', "-DMPIEXEC_NUMPROC_FLAG=%s" % self.cfg['mpiexec_numproc_flag'])
-                    self.cfg.update('configopts', "-DNUMPROC=%s" % self.cfg['mpi_numprocs'])
-                elif self.cfg['runtest']:
-                    raise EasyBuildError("'%s' not found in $PATH", self.cfg['mpiexec'])
-
-
-                self.log.info("Using %s as MPI executable when testing, with numprocs flag '%s' and %s tasks",
-                              self.cfg['mpiexec'], self.cfg['mpiexec_numproc_flag'], self.cfg['mpi_numprocs'])
-
-                # clean up obj dir before reconfiguring
-                shutil.rmtree(os.path.join(self.builddir, 'easybuild_obj'))
-
-                # rebuild/test/install with MPI options
-                super(EB_GROMACS, self).configure_step()
-                super(EB_GROMACS, self).build_step()
-                super(EB_GROMACS, self).test_step()
-                super(EB_GROMACS, self).install_step()
-
-                self.log.info("A full regression test suite is available from the GROMACS web site")
+            if LooseVersion(self.version) < LooseVersion('5.0'):
+                libname = 'libgmx*.%s' % self.libext
+            else:
+                libname = 'libgromacs*.%s' % self.libext
+
+            for libdir in ['lib', 'lib64']:
+                if os.path.exists(os.path.join(self.installdir, libdir)):
+                    for subdir in [libdir, os.path.join(libdir, '*')]:
+                        libpaths = glob.glob(os.path.join(self.installdir, subdir, libname))
+                        if libpaths:
+                            self.lib_subdir = os.path.dirname(libpaths[0])[len(self.installdir)+1:]
+                            self.log.info("Found lib subdirectory that contains %s: %s", libname, self.lib_subdir)
+                            break
+            if not self.lib_subdir:
+                raise EasyBuildError("Failed to determine lib subdirectory in %s", self.installdir)
+
+            # Reset installopts etc for the benefit of the gmxapi extension
+            self.cfg['installopts'] = self.orig_installopts
+
+    def extensions_step(self, fetch=False):
+        """ Custom extensions step, only handle extensions after the last iteration round"""
+        if self.iter_idx < self.variants_to_build - 1:
+            self.log.info("skipping extension step %s", self.iter_idx)
+        else:
+            # Set runtest to None so that the gmxapi extension doesn't try to
+            # run "check" as a command
+            self.cfg['runtest'] = None
```

Comment:
still need to restore original `runtest` value here?

Reply:
Did you miss my question on the previous review of this line?

Repeating:
Do I need to? This is at the last iteration and extensions_step is that last thing done. So nothing else should happen as far as I know.

Reply:
Anyway, done

Reply:
better to do, to avoid introducing bugs later when we make more changes